src/osd/osd_types.cc

```diff
@@ -6178,24 +6178,30 @@ void object_info_t::copy_user_bits(const object_info_t& other)
 
 void object_info_t::encode(ceph::buffer::list& bl, uint64_t features) const
```

Comment:
There are places where we encode object_info_t other than the '_' attr on the object.  Hacking up the encode/decode method here will break those users.  Can we modify crimson to write a different, truncated object_info_t (object_info_no_oid_t) in the attr?  We actually want to do this in classic as well.

Reply:
Added the separate `encode_no_oid()` and `decode_no_oid()` for crimson path. Also use `ceph_abort()` to make sure we won't hit the original denc path in crimson.